dubbo-cluster/src/main/java/org/apache/dubbo/rpc/cluster/router/condition/MultiDestConditionRouter.java

```diff
@@ -0,0 +1,409 @@
+                matcherPair = this.getMatcher(content);
+                condition.put(content, matcherPair);
+            }
+            // The KV part of the condition expression
+            else if ("&".equals(separator)) {
+                if (condition.get(content) == null) {
+                    matcherPair = this.getMatcher(content);
+                    condition.put(content, matcherPair);
+                } else {
+                    matcherPair = condition.get(content);
+                }
+            }
+            // The Value in the KV part.
+            else if ("=".equals(separator)) {
+                if (matcherPair == null) {
+                    throw new ParseException(
+                            "Illegal route rule \"" + rule + "\", The error char '" + separator + "' at index "
+                                    + matcher.start() + " before \"" + content + "\".",
+                            matcher.start());
+                }
+
+                values = matcherPair.getMatches();
+                values.add(content);
+            }
+            // The Value in the KV part.
+            else if ("!=".equals(separator)) {
+                if (matcherPair == null) {
+                    throw new ParseException(
+                            "Illegal route rule \"" + rule + "\", The error char '" + separator + "' at index "
+                                    + matcher.start() + " before \"" + content + "\".",
+                            matcher.start());
+                }
+
+                values = matcherPair.getMismatches();
+                values.add(content);
+            }
+            // The Value in the KV part, if Value have more than one items.
+            else if (",".equals(separator)) { // Should be separated by ','
+                if (values == null || values.isEmpty()) {
+                    throw new ParseException(
+                            "Illegal route rule \"" + rule + "\", The error char '" + separator + "' at index "
+                                    + matcher.start() + " before \"" + content + "\".",
+                            matcher.start());
+                }
+                values.add(content);
+            } else {
+                throw new ParseException(
+                        "Illegal route rule \"" + rule + "\", The error char '" + separator + "' at index "
+                                + matcher.start() + " before \"" + content + "\".",
+                        matcher.start());
+            }
+        }
+        return condition;
+    }
+
+    private ConditionMatcher getMatcher(String key) {
+        for (ConditionMatcherFactory factory : matcherFactories) {
+            if (factory.shouldMatch(key)) {
+                return factory.createMatcher(key, moduleModel);
+            }
+        }
+        return moduleModel
+                .getExtensionLoader(ConditionMatcherFactory.class)
+                .getExtension("param")
```

Comment:
Hard code here. If there only exist **one** implementation, use it directly

Reply:
Here I just kept the same code as v3.0.Let me think about whether there's room for optimization.

Reply:
In the route method, the value in invokers is not modified directly, and there are three cases: 
1.Return invokers directly 
2. return BItlist.empty 
3. The invokers will be cloned and returned

 so I don't think it is necessary to clone first here, just keep the same as v3.0